test/apple/macos-system-helper.test.ts

```diff
@@ -0,0 +1,163 @@
+            encoding: 'utf8',
+          }).trim();
+        }
+
+        // -- Act --
+        const sdkRootDirectoryPath =
+          MacOSSystemHelpers.findSDKRootDirectoryPath();
+
+        // -- Assert --
+        expect(sdkRootDirectoryPath).toBe(sdkPath);
+      },
+    );
+
+    test.runIf(process.platform !== 'darwin')(
+      'should return undefined on non-macOS platforms',
+      () => {
+        // The purpose of this test is to verify that the implementation of findSDKRootDirectoryPath() is still unchanged
+
+        // -- Act --
+        const sdkRootDirectoryPath =
+          MacOSSystemHelpers.findSDKRootDirectoryPath();
+
+        // -- Assert --
+        expect(sdkRootDirectoryPath).toBeUndefined();
+      },
+    );
+  });
+
+  describe('findDeveloperDirectoryPath', () => {
+    test.runIf(process.platform === 'darwin')(
+      'should return the developer directory path',
+      () => {
+        // -- Arrange --
+        let xcodeAppPath: string;
+        if (process.env.XCODE_DEVELOPER_DIR_PATH) {
+          // It is expected that the XCODE_DEVELOPER_DIR_PATH environment variable is set in CI.
+          xcodeAppPath = process.env.XCODE_DEVELOPER_DIR_PATH;
+        } else {
+          // This is a fallback implementation for local development.
+          // It mostly verifies that the implementation of findDeveloperDirectoryPath() is still unchanged,
+          // not that the path is correct.
+          xcodeAppPath = execSync('xcode-select --print-path', {
+            encoding: 'utf8',
+          }).trim();
+        }
+
+        // -- Act --
+        const developerDirectoryPath =
+          MacOSSystemHelpers.findDeveloperDirectoryPath();
+
+        // -- Assert --
+        expect(developerDirectoryPath).toBe(xcodeAppPath);
+      },
+    );
+
+    test.runIf(process.platform !== 'darwin')(
+      'should return undefined on non-macOS platforms',
+      () => {
+        // The purpose of this test is to verify that the implementation of findDeveloperDirectoryPath() is still unchanged
+
+        // -- Act --
+        const developerDirectoryPath =
+          MacOSSystemHelpers.findDeveloperDirectoryPath();
+
+        // -- Assert --
+        expect(developerDirectoryPath).toBeUndefined();
+      },
+    );
+  });
+
+  describe('readXcodeBuildSettings', () => {
+    test.runIf(process.platform === 'darwin')(
+      'should return the build settings',
+      () => {
+        // -- Act --
+        const buildSettings = MacOSSystemHelpers.readXcodeBuildSettings(
+          projectWithSynchronizedFolders,
+        );
+
+        // -- Assert --
+        // The build settings are a massive list of key-value pairs, so we'll just check a few of them
+        // which are relevant for our use cases.
+        expect(buildSettings?.['CONFIGURATION_BUILD_DIR']).toEqual(
+          path.join(
+            appleProjectsPath,
+            'xcode-15-compatible-project/build/Release-unknown',
```

Comment:
`l`: Can't this break easily when it relies on xcode-15?

Reply:
The project directory is called `xcode-15-compatible-project`. It's just a project where I hard-coded the object version number